### DISCUSSION TITLE
Catalog with only one alias

### DISCUSSION THREAD

**Discussion by aghasemi** [jbangdev/jbang#1264] [Ideas] [answered]
Hi,

When a catalog has only one alias (e.g. in a repo dedicated to only one JBang script), is there a shortcut way to install it with jbang, instead of writing something like `jbang app install mycmd@myrepo/mycmd` which has a little bit of redundancy?

Thanks

**Comment by maxandersen** [comment, marked as answer]
No, not currently. 

Not sure what the rule would be but open for suggestions.

**Reply by aghasemi** [reply]
"Naturally", it may be "jbang app install @githubuser/myrepo",  which can then get simpler by removing the @ sign, no?

**Reply by maxandersen** [reply]
And what would it do ?

Pick at random an alias ?

Btw. You do realize you can have a jbang-catalog repo so it just becomes myapp@hithubuser ?

**Reply by quintesse** [reply]
> Pick at random an alias ?

@aghasemi mentions "when a catalog has only one alias". 

And if there are multiple aliases we could probably define a default name, like "main" or "app" or "default" etc.

Thing is that I'm not sure if we could get rid of the `@` anyway. It would look very much like a normal file or alias which would mean that each time somebody mistypes a word we'd be contacting GitHub, GitLab and Bitbucket. That starts to look like a potential DDOS attack to me 😅

**Reply by aghasemi** [reply]
@quintesse Yes, I meant exactly this.

I see your point, then we can just keep @ if it helps. I am not sure if a default name should be there, as it is the name of the command that will be added to the path (no?), hence different for nay script. If needed, we can "assume" that the command name is the name of the script file when no alias name (or a 'default' one) is given.

**Reply by quintesse** [reply]
Aha, I see now, you want to be able to do `jbang app install @somerepo` which would then automatically install the command with the proper name, eg `somecmd`.

It's not much shorter/easier than `jbang app install somecmd@somerepo`, though, especially given the fact you're probably just running it once and never again.

**Reply by aghasemi** [reply]
Fair enough :D
